ffi/diplomat/src/displaynames.rs

```diff
@@ -0,0 +1,141 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+#[diplomat::bridge]
+pub mod ffi {
+    use crate::errors::ffi::ICU4XError;
+    use crate::locale::ffi::ICU4XLocale;
+    use crate::provider::ffi::ICU4XDataProvider;
+    use alloc::boxed::Box;
+    use diplomat_runtime::DiplomatWriteable;
+    #[allow(unused_imports)] // feature-specific
+    use icu_displaynames::options::{DisplayNamesOptions, Fallback, LanguageDisplay};
+    use icu_displaynames::{LanguageDisplayNames, RegionDisplayNames};
+    use writeable::Writeable;
+
+    //  FFI version of `LanguageDisplayNames`.
+    #[diplomat::opaque]
+    #[diplomat::rust_link(icu::displaynames::LanguageDisplayNames, Struct)]
+    pub struct ICU4XLanguageDisplayNames(pub LanguageDisplayNames);
+
+    //  FFI version of `RegionDisplayNames`.
+    #[diplomat::opaque]
+    #[diplomat::rust_link(icu::displaynames::RegionDisplayNames, Struct)]
+    pub struct ICU4XRegionDisplayNames(pub RegionDisplayNames);
+
+    //  FFI version of `DisplayNamesOptions`.
+    #[diplomat::rust_link(icu::displaynames::options::DisplayNamesOptions, Struct)]
+    pub struct ICU4XDisplayNamesOptions {
+        /// The optional formatting style to use for display name.
+        pub style: ICU4XDisplayNamesStyle,
+        /// The fallback return when the system does not have the
+        /// requested display name, defaults to "code".
+        pub fallback: ICU4XDisplayNamesFallback,
+        /// The language display kind, defaults to "dialect".
+        pub language_display: ICU4XLanguageDisplay,
+    }
```

Comment:
Actually, if you're willing to make more changes, I might prefer

```
    pub struct ICU4XDisplayNamesOptions {
        /// The optional formatting style to use for display name.
        pub length: ICU4XDisplayNamesLength,
        /// The fallback return when the system does not have the
        /// requested display name, defaults to "code".
        pub fallback: ICU4XDisplayNamesFallback,
        /// The language display kind, defaults to "dialect".
        pub language_style: ICU4XDisplayNamesLanguageStyle,
    }
```

and the Rust types should probably take those names, too.

This isn't necessarily consistent with 402 but we've had the discussion about "style" versus "length" before and I think we decided that ICU4X should be self-consistent.